software/rasusa/main.nf

```diff
@@ -0,0 +1,43 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process RASUSA {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::rasusa=0.3.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/rasusa:0.3.0--h779adbc_1"
+    } else {
+        container "quay.io/biocontainers/rasusa:0.3.0--h779adbc_1"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    val depth_cutoff
+    val genome_size
+
+    output:
+    tuple val(meta), path('*.fastq.gz'), emit: reads
+    path '*.version.txt'                , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}_${depth_cutoff}X" : "${meta.id}_${depth_cutoff}X"
```

Comment:
```suggestion
    def prefix   = options.suffix ? "${meta.id}.${options.suffix}" : "${meta.id}"
```

Reply:
The suffix should come in via `options.suffix` to keep it completely flexible for developers to name their files however they want.

Reply:
OK, how can I assign depth_cutoff to suffix in a flexible way so that we don't need to define this twice?